graphql_compiler/tests/snapshot_tests/test_orientdb_match_query.py

```diff
@@ -0,0 +1,110 @@
+
+    # We need to preprocess the results to be agnostic of the returned order.
+    # For this we perform the following steps
+    # - convert lists (returned from @fold scopes) to tuples to make them hashable
+    # - convert each row dict to a frozenset of its items
+    # - create a Counter (multi-set) of the row frozensets
+    # - convert the multi-set to a frozenset of its items
+    row_dicts = [row.oRecordData for row in client.command(result.query)]
+    row_dicts_using_tuples = [
+        {
+            column_name: tuple(value) if isinstance(value, list) else value
+            for (column_name, value) in row.items()
+        }
+        for row in row_dicts
+    ]
+    row_frozensets = [frozenset(row_dict.items()) for row_dict in row_dicts_using_tuples]
+    rows_multiset = Counter(row_frozensets)
+    row_counters_frozenset = frozenset(rows_multiset.items())
+
+    return row_counters_frozenset
+
+
+class OrientdbMatchQueryTests(TestCase):
+
+    def setUp(self):
+        """Initialize the test schema once for all tests, and disable max diff limits."""
+        self.maxDiff = None
+        self.schema = get_schema()
+
+    @pytest.mark.usefixtures('graph_client')
+    def test_immediate_output(self):
+        test_data = test_input_data.immediate_output()
+
+        rows = execute_graphql(self.schema, test_data, self.graph_client)
+
+        self.assertMatchSnapshot(rows)
+
+    @pytest.mark.usefixtures('graph_client')
+    def test_traverse_and_output(self):
+        test_data = test_input_data.traverse_and_output()
+
+        rows = execute_graphql(self.schema, test_data, self.graph_client)
+
+        self.assertMatchSnapshot(rows)
+
+    @pytest.mark.usefixtures('graph_client')
+    def test_traverse_filter_and_output(self):
+        test_data = test_input_data.traverse_filter_and_output()
+
+        rows = execute_graphql(self.schema, test_data, self.graph_client)
+
+        self.assertMatchSnapshot(rows)
+
+    @pytest.mark.usefixtures('graph_client')
+    def test_filter_in_optional_block(self):
+        test_data = test_input_data.filter_in_optional_block()
+
+        rows = execute_graphql(self.schema, test_data, self.graph_client)
+
+        self.assertMatchSnapshot(rows)
+
+    @pytest.mark.usefixtures('graph_client')
+    def test_optional_traverse_after_mandatory_traverse(self):
+        test_data = test_input_data.optional_traverse_after_mandatory_traverse()
+
+        rows = execute_graphql(self.schema, test_data, self.graph_client)
+
+        self.assertMatchSnapshot(rows)
+
+    @pytest.mark.usefixtures('graph_client')
+    def test_optional_and_deep_traverse(self):
+        test_data = test_input_data.optional_and_deep_traverse()
+
+        rows = execute_graphql(self.schema, test_data, self.graph_client)
+
+        self.assertMatchSnapshot(rows)
+
+    @pytest.mark.usefixtures('graph_client')
+    def test_fold_on_output_variable(self):
```

Comment:
Don't do this in this PR, but just for context:
- Eventually we'll want these functions to be programmatically generated for each possible test input.
- We should eventually have a way to identify which queries don't actually capture any data, and amend that. Randomly generated data is fine but probably doesn't hit all of the intricacies of the test cases.